### PR TITLE
Refactor invalid operation exception handling

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #80 

WIP for the MVC change.

cc @kichalla @rynowak

### CODE REVIEW

**Comment by jbagga** [comment]
The existing tests in JsonPatchDocumentTest still work as before even after the refactor. We don't expect any change in behavior.

**Comment by jbagga** [comment]
@rynowak Added tests

File: test/Microsoft.AspNetCore.JsonPatch.Test/OperationBaseTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{
+    public class OperationBaseTests
+    {
+        [Theory]
+        [InlineData("add", OperationType.Add)]
+        [InlineData("copy", OperationType.Copy)]
+        [InlineData("move", OperationType.Move)]
+        [InlineData("remove", OperationType.Remove)]
+        [InlineData("replace", OperationType.Replace)]
+        [InlineData("test", OperationType.Test)]
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.JsonPatch.Test/OperationBaseTests.cs]: Include some variations of casing since this is case-insensitive.

**Comment by jbagga** [comment]
I'll merge this after ensuring it doesn't break MVC. Thanks!

**Pull request merged**